src/main/java/org/jenkinsci/plugins/oic/OicSecurityRealm.java

```diff
@@ -1394,6 +1409,25 @@ public boolean handleTokenExpiration(HttpServletRequest httpRequest, HttpServlet
 
         User user = User.get2(authentication);
 
+        if (isTraditionalApiTokenAccessEnabled()) {
+            // check if this is a valid api token based request
+            if (httpRequest.getHeader("Authorization") != null) {
+                String token = new String(
+                                Base64.getDecoder()
+                                        .decode(httpRequest
+                                                .getHeader("Authorization")
+                                                .substring(6)),
+                                StandardCharsets.UTF_8)
+                        .split(":")[1];
```

Comment:
Isn't there any utility in core to do this?

Reply:
did not find anything with a mild search, happy to change it if you can point me at the code.